src/dataframe/dataframe.jl

```diff
@@ -763,8 +763,47 @@ end
 ##
 ##############################################################################
 
+"""
+    deletecols!(df::DataFrame, ind)
+
+Delete columns specified by `ind` from a `DataFrame` df in place.
+Returns the modified `DataFrame`.
+
+Argument `ind` can be any index that is allowed for column indexing of
+a `DataFrame` provided that the columns requested to be removed are unique.
+
+### Examples
+
+```jldoctest
+julia> d = DataFrame(a=1:3, b=4:6)
+3×2 DataFrame
+│ Row │ a     │ b     │
+│     │ Int64 │ Int64 │
+├─────┼───────┼───────┤
+│ 1   │ 1     │ 4     │
+│ 2   │ 2     │ 5     │
+│ 3   │ 3     │ 6     │
+
+julia> deletecols!(d, 1)
+3×1 DataFrame
+│ Row │ b     │
+│     │ Int64 │
+├─────┼───────┤
+│ 1   │ 4     │
+│ 2   │ 5     │
+│ 3   │ 6     │
+```
+
+"""
 function deletecols!(df::DataFrame, inds::Vector{Int})
-    for ind in sort(inds, rev = true)
+    sorted_inds = sort(inds, rev=true)
+    for i in 2:length(sorted_inds)
+        if sorted_inds[i] == sorted_inds[i-1]
+            throw(ArgumentError("Duplicate values in inds found at positions" *
```

Comment:
Why not call `unique!` instead?

Reply:
The performance impact should be absolutely negligible, and it makes things simpler for us and less annoying for users.

Reply:
But do you mean that you want to allow duplicates and drop them?
I would prefer to throw an error on duplicate.

Reply:
`deleteat!` in Base throws an error on duplicates (and thus so does `deleterows!`). So I implemented this to be consistent with that function. I can drop it and document that we ignore duplicates if you prefer.

Reply:
OK, that's fine.